### ISSUE TITLE
block-scoped-var behavior in 0.16.0 is different from 0.15.0

### ISSUE THREAD

**Issue by shinnn** [eslint/eslint#1969]
For example, when I lint `file.js` like this,

```js
"use strict";

var v = 1;

module.exports = function() {
  return v;
};
```

With eslint 0.15.1,

```sh
$ eslint file.js --env=node --rule=block-scoped-var:2
```

It shows no errors.

But, with eslint 0.16.0,

```sh
$ eslint file.js --env=node --rule=block-scoped-var:2

file.js
  6:9  error  "v" used outside of binding context  block-scoped-var
```

it shows an error.

I think the previous behavior of `block-scoped-var` rule is valid for `node` environement.

**Comment by nzakas** [comment]
Must have something to do with the node scope changes in 0.16.0 since there were only small changes in the rule.

**Comment by shinnn** [comment]
When `ecmaFeatures.globalReturn` is `false`, v0.16.0 works fine, too.

```
env:
  node: true
ecmaFeatures:
  globalReturn: false
rules:
  block-scoped-var: 2
```

[This change](https://github.com/eslint/eslint/commit/e5aea7f92bd6dfe67ca8f6943ae090a72b4115f5#diff-b6559e6b095a292d244b6b7bae986291R649) seems to cause this bug.

**Comment by ryanzec** [comment]
I was also experiencing issues with required modules (using browserify) showing this error:

```
 3:17  error  "storeGenerator" used outside of binding context  block-scoped-var
```

Which seemed incorrect to me since the code is:

```javascript
var storeGenerator = require('../core/store-generator');

module.exports = storeGenerator({
  getMenu: function menuStoreGetMenu() {
  //...
```

Setting `ecmaFeatures.globalReturn` to `false` fixed this issue for me too.

**Comment by nzakas** [comment]
Might be a bug with escope, looking into it.

**Comment by shinnn** [comment]
v0.16.1 works fine for me. Thanks, @nzakas!

**Comment by eTorAken** [comment]
I got this bug back in `v0.18.0`. I tried with `ecmaFeatures.globalReturn` to `false` without any change.

**Comment by nzakas** [comment]
Please file a new issue and include repro steps.
